x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/result_settings_table/field_number.tsx

```diff
@@ -43,11 +43,10 @@ const handleFieldNumberBlur = (
   clearAction: (fieldName: string) => void
 ) => {
   return (e: FocusEvent<HTMLInputElement>) => {
-    const value = parseInt(e.target.value, 10);
-    const fieldValue = Math.min(
-      SIZE_FIELD_MAXIMUM,
-      Math.max(SIZE_FIELD_MINIMUM, isNaN(value) ? 0 : value)
-    );
+    let fieldValue = parseInt(e.target.value, 10);
+    if (!isNaN(fieldValue)) {
+      fieldValue = Math.min(SIZE_FIELD_MAXIMUM, Math.max(SIZE_FIELD_MINIMUM, fieldValue));
+    }
```

Comment:
Honestly, I think there's something to be said here for @byronhulcher's suggestion in Slack about removing the `handleFieldNumberBlur` magic and letting the returned errors do the work for us. This is because a user can still do this:

![range](https://user-images.githubusercontent.com/549407/124817416-a11d5880-df1e-11eb-994e-f65b8a6c3bfd.gif)

Because the use never left the input and triggered on blur, the validation never kicked in - but the onChange event still fired and sent an XHR request with an error response. TBH, it might be just easier to fall back to the error response 🤷 

That being said, this is a patch fix for now for 7.14 so this PR primarily just focuses on getting this working

Reply:
I'd be fine with that.

I *think* the blur logic is in there for 2 reasons (carried over from ent-search):

- EuiFieldNumber prevents users from entering non-numeric values except for like `e`, or something like that.
- Apply a min and max value automatically.

The logic is super hairy though. I think it'd be fine to fall back to the error message.